plugins/restful/RestfulDataProviderDbQuery.php

```diff
@@ -198,6 +198,16 @@ protected function queryForListFilter(\SelectQuery $query) {
     foreach ($this->parseRequestForListFilter() as $filter) {
       if (in_array(strtoupper($filter['operator'][0]), array('IN', 'NOT IN', 'BETWEEN'))) {
         $column_name = $this->getPropertyColumnForQuery($public_fields[$filter['public_field']]);
+        if (is_array($filter['value']) && empty($filter['value'])) {
+          if (strtoupper($filter['operator'][0]) == 'NOT IN') {
+            // Skip filtering by an empty value when operator is 'NOT IN',
+            // since it throws an SQL error.
+            continue;
+          }
+          // Since Drupal doesn't know how to handle an empty array within a
+          // condition we add the `NULL` as an element to the array.
```

Comment:
So the idea here is to make sure we get no results back in such a case?

Reply:
I think yes, if someone is trying to filter by a value that is actually an empty array, it means by `NULL`.
At least as I see it.
I've split the `NOT IN` case since it's really a different scenario and when you pass the same `[NULL]` with `NOT IN` you also get no results, which is wrong...